src/ripple/rpc/Yield.h

```diff
@@ -27,19 +28,26 @@
 
 namespace ripple {
 
+class JobQueue;
 class Section;
 
 namespace RPC {
 
-/** Yield is a generic placeholder for a function that yields control of
-    execution - perhaps to another coroutine.
+/** See the README.md in this directory for more information about how
+    the RPC yield mechanism works.
+ */
 
-    When code calls Yield, it might block for an indeterminate period of time.
+/** Callback: do something and eventually return. */
+using Callback = std::function <void ()>;
```

Comment:
The state of all local variables are preserved through boost context magic.  You don't need to do anything special - the code looks sequential to you:

    auto foo = expensiveComputation();
    suspend (myContinuation);  // run on jobqueue, wait for condition, etc.
    makeResponse (foo);   // foo is still good.